extensions/wikia/AdEngine/AdEngine2ContextService.class.php

```diff
@@ -17,7 +17,9 @@ public function getContext( Title $title, $skinName ) {
 			$adPageTypeService = new AdEngine2PageTypeService();
 			$wikiaPageType = new WikiaPageType();
 			$pageType = $wikiaPageType->getPageType();
-			$hasFeaturedVideo = !empty( $wg->EnableArticleFeaturedVideo ) && ArticleVideoContext::isFeaturedVideoEmbedded( $title->getPrefixedDBkey() );
+			$hasFeaturedVideo = !empty( $wg->EnableArticleFeaturedVideo )
+				&& !empty( $title->getArticleID() )
```

Comment:
I'd check it inside isFeaturedVideoEmbedded method

Reply:
I'd like to leave it here. It should be responsibility of client of given method to provide correct data to the method rather than doing checks within the method. This way logic within the method is clearer and focused on problem it solves. Also we use type hints for parameters, so if null is provided instead of articleId, then php will throw warning